### ISSUE TITLE
[BUG] Restoring a bottle from a backup creates the wrong directory and doesn't work

### ISSUE THREAD

**Issue by paddylandau** [bottlesdevs/Bottles#913]
## Describe the bug

When restoring a backup of a bottle, the backup creates an entire folder hierarchy instead of the correct hierarchy. Even when correcting for that, the restored bottle still doesn't work.

Manually backing up and restoring, however, does work.

This [has been discussed](https://forums.usebottles.com/d/47-export-quirk-and-restore-failure) in the Bottles forums, and I was asked to report this here.

### Bottles installation

- Installed via Flatpak. You will see in the explanation below that the full path is in the Flatpak area.
- Version 2021.12.28-treviso (64-bit)

### Desktop

- OS: Ubuntu
- Version: 20.04
- Display server: Gnome

## To Reproduce

1. Go to a bottle.
1. Create a full archive using the down-arrow ("Make a backup of this bottle") > Full archive.
1. Wait for the export to complete ~~(there is no indication of this, so you have to watch the file size and the CPU activity, and assume that it's finished when the file no longer changes size, and the CPU settles down)~~ (just look the operations menu from the topbar _Mirko).
1. Delete the bottle.
1. Restore the bottle: Import/export > Import Bottles backup > Full archive.

## Expected behavior

It is expected that the Bottle will be restored fully as if it hadn't been deleted.

### Incorrect file structure

The Bottle is completely non-functional ([screenshot](https://pasteboard.co/9lYJ1tF8gnrl.png)). The reason is that the full hierarchy of the backed-up folder is recreated. For the purposes of explanation, I'll use my bottle, which is named Imported_Quicken.

The original looks like this:

```
/home/paddy/.var/app/com.usebottles.bottles/data/bottles/bottles/Imported_Quicken/
```
with the bottle files within that folder.

However, after restoring the bottle, the file hierarchy looks like this:

```
/home/paddy/.var/app/com.usebottles.bottles/data/bottles/bottles/Imported_Quicken/home/paddy/.var/app/com.usebottles.bottles/data/bottles/bottles/Imported_Quicken/
```
with the bottle files within that folder.

Notice how the entire structure has been duplicated.

### Manual fix (doesn't work)

If I close Bottles, manually correct this structure, and reopen Bottles, it no longer complains. However, the bottle fails to run correctly. Below is the output from running it from the command line.

```
$ flatpak run com.usebottles.bottles --bottle=Imported_Quicken --lnk=${HOME}/.var/app/com.usebottles.bottles/data/bottles/bottles/Imported_Quicken/drive_c/users/paddy/'Start Menu'/Programs/'Quicken Deluxe 2000'/'Quicken Deluxe 2000.lnk'
Gtk-Message: 17:46:34.985: Failed to load module "canberra-gtk-module"
(17:46:35) INFO Performing Bottles checks... 
(17:46:35) INFO DXVKs found:
	 - dxvk-1.9.2
 
(17:46:35) INFO VKD3Ds found:
	 - vkd3d-proton-2.5
 
(17:46:35) INFO NVAPIs found:
	 - dxvk-nvapi-v0.5.1
 
(17:46:35) INFO Runners found:
	 - caffe-6.23
 
(17:46:35) INFO Bottles found:
	 - Imported_Quicken
 
(17:46:35) INFO Running link file on the bottle… 
wineserver: using server-side synchronization.
wine: could not open working directory L"C:\\windows\\system32\\", starting in the Windows directory.
002c:err:wineboot:main Cannot set the dir to L"C:\\windows" (2)
002c:err:wineboot:process_run_key Error running cmd L"C:\\windows\\system32\\winemenubuilder.exe -a -r" (2).
wine: could not open working directory L"C:\\windows\\system32\\", starting in the Windows directory.
002c:err:shell:SHGetFolderPathAndSubDirW Failed to create directory L"C:\\users\\paddy\\Application Data".
002c:err:shell:SHGetFolderPathAndSubDirW Failed to create directory L"C:\\users\\paddy".
002c:err:shell:SHGetFolderPathAndSubDirW Failed to create directory L"C:\\users\\paddy\\Local Settings\\Application Data".
002c:err:shell:SHGetFolderPathAndSubDirW Failed to create directory L"C:\\users\\paddy\\Local Settings\\Temporary Internet Files".
002c:err:wininet:cache_containers_init Couldn't get path for default container 0
002c:err:shell:SHGetFolderPathAndSubDirW Failed to create directory L"C:\\users\\paddy\\Local Settings\\History".
002c:err:wininet:cache_containers_init Couldn't get path for default container 1
002c:err:shell:SHGetFolderPathAndSubDirW Failed to create directory L"C:\\users\\paddy\\Cookies".
002c:err:wininet:cache_containers_init Couldn't get path for default container 2
wine: could not load kernel32.dll, status c0000135
```

## Workaround

I have found a successful workaround, which I am using for now.

1. Manually back up the entire bottle, i.e. the folder and its contents.
1. To restore to a fresh installation of Bottles, simply restore the folder. Bottles recognises it, and the restored bottle works correctly.

I'm not a programmer, but if I can help with diagnosing or testing, please ask.

_Thank you_

**Comment by mirkobrombin** [comment]
Thanks for reporting

**Comment by ourichermath** [comment]
Did you actually get Quicken working in a Bottle?  I've got the latest version and couldn't get it to run at all (ditto for regular Wine, too).  Are you running the current version of Quicken or an old, legacy version?

**Comment by paddylandau** [comment]
> Did you actually get Quicken working in a Bottle? I've got the latest version and couldn't get it to run at all (ditto for regular Wine, too). Are you running the current version of Quicken or an old, legacy version?

Mine is a very old version from 2000. It works in WINE (via PlayOnLinux). But, I have to back up the WINE and PlayOnLinux setup, because it's no longer possible to install it from scratch — it requires letting the app phone (yes, phone via a modem) a number that doesn't even exist any more.

I did get it working in Bottles for a while by importing the setup from WINE, but then after a Bottles update, it stopped working. I've returned to WINE because it's easier for me :)
